### PR TITLE
Add QuarkusApplication javadoc

### LINKED ISSUE #34463: QuarkusApplication Simple JavaDoc
### Description

There will be times that "new" developers in using Quarkus will be implementing the FunctionalInterface, `QuarkusApplication`. Seeing in the Quarkus Guides, when you put logic in the Main class, you should implement this interface.

Proposing to have just a simple class level documentation on what is `QuarkusApplication`. Since it is also a public visibility class exported API to the users.

### Implementation ideas

Add a class-level javadoc for `QuarkusApplication` interface.

Will open a PR on this one.

### PR DESCRIPTION
**Pull request by rjtmahinay**
* Fix #34463

### CODE REVIEW

**Comment by rjtmahinay** [comment]
For review @geoand

File: core/runtime/src/main/java/io/quarkus/runtime/QuarkusApplication.java [hunk 1]
```diff
@@ -1,6 +1,17 @@
 package io.quarkus.runtime;
 
+/**
+ * Defines a class as a QuarkusApplication ran by {@link Quarkus}. This is usually used for command mode applications
```
Comment (geoand) [inline core/runtime/src/main/java/io/quarkus/runtime/QuarkusApplication.java:4]: I think the first sentence is redundant

File: core/runtime/src/main/java/io/quarkus/runtime/QuarkusApplication.java [hunk 2]
```diff
@@ -1,6 +1,17 @@
...
+ * with startup logic. The logic is executed inside {@link QuarkusApplication#run} method before
+ * the main application exits.
+ */
 public interface QuarkusApplication {
 
+    /**
+     * Method to run a QuarkusApplication.
```
Comment (geoand) [inline core/runtime/src/main/java/io/quarkus/runtime/QuarkusApplication.java:11]: I don't think this method really needs any Javadoc to be honest

**Review by geoand — COMMENTED** [review]
Thanks for this.

I added a couple of comments

File: core/runtime/src/main/java/io/quarkus/runtime/QuarkusApplication.java (hunk 1 above)
Comment (rjtmahinay) [reply core/runtime/src/main/java/io/quarkus/runtime/QuarkusApplication.java]: @geoand Kinda make sense. Changes already reflected

File: core/runtime/src/main/java/io/quarkus/runtime/QuarkusApplication.java (hunk 2 above)
Comment (rjtmahinay) [reply core/runtime/src/main/java/io/quarkus/runtime/QuarkusApplication.java]: For now let's leave it undocumented. Maybe in the future some new customers (framework users) will ask what's the usage of this run function in a method level for their convenience.

**Review by geoand — APPROVED** [review]
Thanks!

**Pull request merged**